ui/src/components/role/AddRole.js

```diff
@@ -171,6 +176,13 @@ class AddRole extends React.Component {
         };
     }
 
+    toggleAuditEnabled() {
+        this.state.role.auditEnabled = !this.state.role.auditEnabled;
```

Comment:
you shouldn't set the value of this.state.role.auditEnabled it destroy the immutable.
you should extract the value to a different var.
look immer library it should help you.

Reply:
Hey @royelbaum thank you for the code review!
I went ahead and integrated immer lib in my code.